### PR TITLE
Add custom category and rule for naming conventions

### LINKED ISSUE #19: Allow enforcement of custom naming conventions
While we'll want to enforce snake_case naming of rules and vars by default, it'd be nice if we provided an optional rule to enforce custom naming conventions. This could be provided in the form of a regex, that would be checked against all rule names and vars.

### PR DESCRIPTION
**Pull request by anderseknert**
This should hopefully help people satisfy the most common organizational requirements without having to write their own rules. First off is the naming-convention rule, with more to follow later.

Fixes #19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,7 @@ Documentation:	https://github.com/StyraInc/regal/blob/main/docs/rules/style/use-
 Regal comes with a set of built-in rules, grouped by category.
 
 - **bugs**: Common mistakes, potential bugs and inefficiencies in Rego policies.
+- **custom**: Custom, or organizational rules, where you decide what to enforce.
```
Comment (charlieegan3) [inline README.md:132]:
```suggestion
- **custom**: Custom, rules where enforcement can be adjusted to match your organization or team preferences.
```

File: README.md [hunk 2]
```diff
@@ -175,12 +177,24 @@ The following rules are currently available:
...
+By default, all rules except for those in the `custom` category are currently **enabled**.
 
 If you'd like to see more rules, please [open an issue](https://github.com/StyraInc/regal/issues) for your feature
 request, or better yet, submit a PR! See the [custom rules](/docs/custom-rules.md) page for more information on how to
 develop your own rules, for yourself or for inclusion in Regal.
 
+### Custom Rules
+
+The `custom` category is a special one, as the rules in this category allow you to enforce rules that are specific to
+your project, team or organization. This typically include things like naming conventions, where you might want to
```
Comment (charlieegan3) [inline README.md:189]:
```suggestion
your project, team or organization. This typically includes things like naming conventions, where you might want to
```

File: README.md [hunk 3]
```diff
@@ -175,12 +177,24 @@ The following rules are currently available:
...
+ensure that e.g. all package names adhere to an organizational standard, like having a prefix matching the
```
Comment (charlieegan3) [inline README.md:190]:
```suggestion
ensure, for example, that all package names adhere to an organizational standard, like having a prefix matching the
```

File: e2e/cli_test.go [hunk 4]
```diff
@@ -5,6 +5,7 @@ package e2e
 import (
 	"bytes"
 	"encoding/json"
+	"github.com/styrainc/regal/internal/util"
```
Comment (charlieegan3) [inline e2e/cli_test.go:8]: nit: import order

**Review by charlieegan3 — APPROVED** [review]
This looks really good!

File: e2e/cli_test.go (hunk 4 above)
Comment (anderseknert) [reply e2e/cli_test.go]: Interesting. I wonder why it's not caught by `golangci-lint run ./...` 🤔 But fixed now either way!

File: README.md (hunk 3 above)
Comment (anderseknert) [reply README.md]: Fixed!

File: README.md (hunk 2 above)
Comment (anderseknert) [reply README.md]: Thanks! Fixed.

File: README.md (hunk 1 above)
Comment (anderseknert) [reply README.md]: Thanks! I shortened it down to just "your preferences" so that we wouldn't have a linke break in that list.

**Pull request merged**